diff --git a/import_export/fields.py b/import_export/fields.py
index 23839f7..30dab31 100644
--- a/import_export/fields.py
+++ b/import_export/fields.py
@@ -63,6 +63,8 @@ class Field:
         self.saves_null_values = saves_null_values
         self.dehydrate_method = dehydrate_method
         self.m2m_add = m2m_add
+        self._row_id = None
+        self._cleaned_value = None
 
     def __repr__(self):
         """
@@ -78,6 +80,11 @@ class Field:
         Translates the value stored in the imported datasource to an
         appropriate Python object and returns it.
         """
+        current_row_id = id(row)
+
+        if self._row_id == current_row_id:
+            return self._cleaned_value
+
         try:
             value = row[self.column_name]
         except KeyError:
@@ -90,9 +97,12 @@ class Field:
 
         if value in self.empty_values and self.default != NOT_PROVIDED:
             if callable(self.default):
-                return self.default()
-            return self.default
+                value = self.default()
+            else:
+                value = self.default
 
+        self._row_id = current_row_id
+        self._cleaned_value = value
         return value
 
     def get_value(self, instance):
