diff --git a/import_export/formats/base_formats.py b/import_export/formats/base_formats.py
index efb5fc9..8042371 100644
--- a/import_export/formats/base_formats.py
+++ b/import_export/formats/base_formats.py
@@ -13,22 +13,10 @@ from tablib.formats import registry
 logger = logging.getLogger(__name__)
 
 
-class Format:
+class FilePropertiesMixin:
     def get_title(self):
         return type(self)
 
-    def create_dataset(self, in_stream):
-        """
-        Create dataset from given string.
-        """
-        raise NotImplementedError()
-
-    def export_data(self, dataset, **kwargs):
-        """
-        Returns format representation for given dataset.
-        """
-        raise NotImplementedError()
-
     def is_binary(self):
         """
         Returns if this format is binary.
@@ -52,6 +40,20 @@ class Format:
         # https://www.iana.org/assignments/media-types/media-types.xhtml
         return "application/octet-stream"
 
+
+class DataSerializationMixin:
+    def create_dataset(self, in_stream):
+        """
+        Create dataset from given string.
+        """
+        raise NotImplementedError()
+
+    def export_data(self, dataset, **kwargs):
+        """
+        Returns format representation for given dataset.
+        """
+        raise NotImplementedError()
+
     @classmethod
     def is_available(cls):
         return True
@@ -63,6 +65,10 @@ class Format:
         return False
 
 
+class Format(FilePropertiesMixin, DataSerializationMixin):
+    pass
+
+
 class TablibFormat(Format):
     TABLIB_MODULE = None
     CONTENT_TYPE = "application/octet-stream"
