diff --git a/import_export/admin.py b/import_export/admin.py
index 4fdb6ed..3accaf2 100644
--- a/import_export/admin.py
+++ b/import_export/admin.py
@@ -730,10 +730,13 @@ class ExportMixin(BaseExportMixin, ImportExportMixinBase):
             raise PermissionDenied
 
         form_type = self.get_export_form_class()
-        formats = self.get_export_formats()
+        formats = [
+            file_format if not isinstance(file_format, type) else file_format()
+            for file_format in self.get_export_formats()
+        ]
         queryset = self.get_export_queryset(request)
         if self.is_skip_export_form_enabled():
-            return self._do_file_export(formats[0](), request, queryset)
+            return self._do_file_export(formats[0], request, queryset)
 
         form = form_type(
             formats,
@@ -749,7 +752,7 @@ class ExportMixin(BaseExportMixin, ImportExportMixinBase):
                 choices=[(pk, pk) for pk in queryset.values_list("pk", flat=True)],
             )
         if form.is_valid():
-            file_format = formats[int(form.cleaned_data["format"])]()
+            file_format = form.get_selected_format()
 
             if "export_items" in form.changed_data:
                 # this request has arisen from an Admin UI action
@@ -870,13 +873,15 @@ class ExportActionMixin(ExportMixin):
         """
         Action runs on POST from instance action menu (if enabled).
         """
-        formats = self.get_export_formats()
+        formats = [
+            file_format if not isinstance(file_format, type) else file_format()
+            for file_format in self.get_export_formats()
+        ]
         if self.is_skip_export_form_from_action_enabled():
-            file_format = formats[0]()
+            file_format = formats[0]
             return self._do_file_export(file_format, request, queryset)
 
         form_type = self.get_export_form_class()
-        formats = self.get_export_formats()
         export_items = list(queryset.values_list("pk", flat=True))
         form = form_type(
             formats=formats,
diff --git a/import_export/forms.py b/import_export/forms.py
index 6e68e4f..c766f16 100644
--- a/import_export/forms.py
+++ b/import_export/forms.py
@@ -44,8 +44,10 @@ class ImportExportFormBase(FieldNamePrefixMixin, forms.Form):
 
     def __init__(self, formats, resources, **kwargs):
         super().__init__(**kwargs)
-        self._init_resources(resources)
-        self._init_formats(formats)
+        self.formats = list(formats)
+        self.resources = list(resources)
+        self._init_resources(self.resources)
+        self._init_formats(self.formats)
 
     def _init_resources(self, resources):
         if not resources:
@@ -63,17 +65,32 @@ class ImportExportFormBase(FieldNamePrefixMixin, forms.Form):
         if not formats:
             raise ValueError("invalid formats list")
 
-        choices = [(str(i), f().get_title()) for i, f in enumerate(formats)]
-        if len(formats) == 1:
+        self.formats = [
+            file_format if not isinstance(file_format, type) else file_format()
+            for file_format in formats
+        ]
+        choices = [
+            (str(i), file_format.get_title())
+            for i, file_format in enumerate(self.formats)
+        ]
+        if len(self.formats) == 1:
             field = self.fields["format"]
-            field.value = formats[0]().get_title()
+            field.value = self.formats[0].get_title()
             field.initial = 0
             field.widget.attrs["readonly"] = True
-        if len(formats) > 1:
+        if len(self.formats) > 1:
             choices.insert(0, ("", "---"))
 
         self.fields["format"].choices = choices
 
+    def get_selected_format(self):
+        if not getattr(self, "cleaned_data", None):
+            raise forms.ValidationError(
+                _("Form is not validated, call `is_valid` first")
+            )
+
+        return self.formats[int(self.cleaned_data["format"])]
+
 
 class ImportForm(ImportExportFormBase):
     import_file = forms.FileField(label=_("File to import"))
diff --git a/import_export/mixins.py b/import_export/mixins.py
index 72f58bb..dd9176e 100644
--- a/import_export/mixins.py
+++ b/import_export/mixins.py
@@ -140,7 +140,16 @@ class BaseExportMixin(BaseImportExportMixin):
         """
         Returns available export formats.
         """
-        return [f for f in self.export_formats if f().can_export()]
+        formats = []
+        for format_class in self.export_formats:
+            file_format = format_class()
+            if file_format.can_export():
+                try:
+                    file_format.__name__ = format_class.__name__
+                except AttributeError:
+                    pass
+                formats.append(file_format)
+        return formats
 
     def get_export_resource_classes(self, request):
         """
