diff --git a/import_export/admin.py b/import_export/admin.py
index 4fdb6ed..7ae5344 100644
--- a/import_export/admin.py
+++ b/import_export/admin.py
@@ -733,7 +733,7 @@ class ExportMixin(BaseExportMixin, ImportExportMixinBase):
         formats = self.get_export_formats()
         queryset = self.get_export_queryset(request)
         if self.is_skip_export_form_enabled():
-            return self._do_file_export(formats[0](), request, queryset)
+            return self._do_file_export(formats[0], request, queryset)
 
         form = form_type(
             formats,
@@ -749,7 +749,7 @@ class ExportMixin(BaseExportMixin, ImportExportMixinBase):
                 choices=[(pk, pk) for pk in queryset.values_list("pk", flat=True)],
             )
         if form.is_valid():
-            file_format = formats[int(form.cleaned_data["format"])]()
+            file_format = formats[int(form.cleaned_data["format"])]
 
             if "export_items" in form.changed_data:
                 # this request has arisen from an Admin UI action
@@ -872,11 +872,10 @@ class ExportActionMixin(ExportMixin):
         """
         formats = self.get_export_formats()
         if self.is_skip_export_form_from_action_enabled():
-            file_format = formats[0]()
+            file_format = formats[0]
             return self._do_file_export(file_format, request, queryset)
 
         form_type = self.get_export_form_class()
-        formats = self.get_export_formats()
         export_items = list(queryset.values_list("pk", flat=True))
         form = form_type(
             formats=formats,
diff --git a/import_export/forms.py b/import_export/forms.py
index 6e68e4f..23c72d0 100644
--- a/import_export/forms.py
+++ b/import_export/forms.py
@@ -63,10 +63,16 @@ class ImportExportFormBase(FieldNamePrefixMixin, forms.Form):
         if not formats:
             raise ValueError("invalid formats list")
 
-        choices = [(str(i), f().get_title()) for i, f in enumerate(formats)]
+        choices = []
+        for i, f in enumerate(formats):
+            # Maintain compatibility with `ImportForm` which still passes classes
+            title = f.get_title() if not isinstance(f, type) else f().get_title()
+            choices.append((str(i), title))
+            
         if len(formats) == 1:
             field = self.fields["format"]
-            field.value = formats[0]().get_title()
+            f = formats[0]
+            field.value = f.get_title() if not isinstance(f, type) else f().get_title()
             field.initial = 0
             field.widget.attrs["readonly"] = True
         if len(formats) > 1:
diff --git a/import_export/mixins.py b/import_export/mixins.py
index 72f58bb..a78ed1b 100644
--- a/import_export/mixins.py
+++ b/import_export/mixins.py
@@ -140,7 +140,7 @@ class BaseExportMixin(BaseImportExportMixin):
         """
         Returns available export formats.
         """
-        return [f for f in self.export_formats if f().can_export()]
+        return [f for f in [fmt() for fmt in self.export_formats] if f.can_export()]
 
     def get_export_resource_classes(self, request):
         """
diff --git a/tests/core/tests/test_mixins.py b/tests/core/tests/test_mixins.py
index 7d3a15d..cefae2e 100644
--- a/tests/core/tests/test_mixins.py
+++ b/tests/core/tests/test_mixins.py
@@ -179,7 +179,7 @@ class BaseExportMixinTest(TestCase):
 
         formats = m.get_export_formats()
         self.assertEqual(1, len(formats))
-        self.assertEqual("CanExportFormat", formats[0].__name__)
+        self.assertEqual("CanExportFormat", formats[0].__class__.__name__)
 
 
 class ExportMixinTest(TestCase):
